match/lib/match/change_password.rb

```diff
@@ -1,31 +1,44 @@
 require_relative 'module'
-require_relative 'encrypt'
-require_relative 'git_helper'
+
+require_relative 'storage'
+require_relative 'encryption'
 
 module Match
   # These functions should only be used while in (UI.) interactive mode
   class ChangePassword
-    def self.update(params: nil, from: nil, to: nil)
+    def self.update(params: nil)
       ensure_ui_interactive
-      to ||= ChangePassword.ask_password(message: "New passphrase for Git Repo: ", confirm: false)
-      from ||= ChangePassword.ask_password(message: "Old passphrase for Git Repo: ", confirm: true)
-      GitHelper.clear_changes
-      workspace = GitHelper.clone(params[:git_url],
-                                  params[:shallow_clone],
-                                  manual_password: from,
-                                  skip_docs: params[:skip_docs],
-                                  branch: params[:git_branch],
-                                  git_full_name: params[:git_full_name],
-                                  git_user_email: params[:git_user_email],
-                                  clone_branch_directly: params[:clone_branch_directly])
-      Encrypt.new.clear_password(params[:git_url])
-      Encrypt.new.store_password(params[:git_url], to)
+
+      to ||= ChangePassword.ask_password(message: "New passphrase for Git Repo: ", confirm: true)
+
+      # Choose the right storage and encryption implementations
+      storage = Storage::Interface.storage_class_for_storage_mode(params[:storage_mode]).new
+
+      storage.configure(git_url: params[:git_url],
+                    shallow_clone: params[:shallow_clone],
+                    skip_docs: params[:skip_docs],
+                    branch: params[:git_branch],
+                    git_full_name: params[:git_full_name],
+                    git_user_email: params[:git_user_email],
+                    clone_branch_directly: params[:clone_branch_directly])
+      storage.download
+
+      encryption = Encryption::Interface.encryption_class_for_storage_mode(params[:storage_mode]).new(
+        git_url: storage.git_url,
+        working_directory: storage.working_directory
+      )
+      encryption.decrypt_files
+
+      encryption.clear_password
+      encryption.store_password(to)
```

Comment:
These two aren't defined on the interface. I think it makes sense to have one public call `change_password`. I'm not sure storing a password makes sense for all encryption interfaces, so I would also add a `has_local_password?` method to the interface. Then this action should have an early return for encryption interfaces that don't save the password to the local keychain.